python/kfserving/kfserving/kfmodels/sklearn.py

```diff
@@ -16,27 +16,40 @@
 import joblib
 import numpy as np
 import os
-from typing import List, Dict
+from typing import Dict
+from kfserving.kfmodels.kfmodel_types import KFModelTypes, MODEL_EXTENSIONS
 
 MODEL_BASENAME = "model"
-MODEL_EXTENSIONS = [".joblib", ".pkl", ".pickle"]
+MODEL_EXTENSIONS = MODEL_EXTENSIONS[KFModelTypes.Sklearn]
 
 
-class SKLearnModel(kfserving.KFModel): #pylint:disable=c-extension-no-member
+class SKLearnModel(kfserving.KFModel):  # pylint:disable=c-extension-no-member
     def __init__(self, name: str, model_dir: str):
         super().__init__(name)
         self.name = name
+        self.full_model_path = ""
         self.model_dir = model_dir
         self.ready = False
+        self._model = None
 
-    def load(self):
+    def load_from_model_dir(self):
         model_path = kfserving.Storage.download(self.model_dir)
```

Comment:
Do we still need to download model? I think the model agent will handle the download from bcs/gcs and the model server only need to load from local file system.

Reply:
We still need to use storage initializer for the single mode, let's keep this logic for single model inference service.
The storage initializer download files 
https://github.com/kubeflow/kfserving/blob/master/python/storage-initializer/scripts/initializer-entrypoint

The load_from_model_dir will not download again, it does some sanity check and create symlink